code/modules/projectiles/guns/misc/blastcannon.dm

```diff
@@ -47,54 +61,58 @@
 	return ..()
 
 /obj/item/gun/blastcannon/update_icon_state()
-	if(bomb)
-		icon_state = icon_state_loaded
-	else
-		icon_state = initial(icon_state)
-
-/obj/item/gun/blastcannon/attackby(obj/O, mob/user)
-	if(istype(O, /obj/item/transfer_valve))
-		var/obj/item/transfer_valve/T = O
-		if(!T.tank_one || !T.tank_two)
-			to_chat(user, "<span class='warning'>What good would an incomplete bomb do?</span>")
-			return FALSE
-		if(!user.transferItemToLoc(T, src))
-			to_chat(user, "<span class='warning'>[T] seems to be stuck to your hand!</span>")
-			return FALSE
-		user.visible_message("<span class='warning'>[user] attaches [T] to [src]!</span>")
-		bomb = T
-		name = "blast cannon"
-		desc = "A makeshift device used to concentrate a bomb's blast energy to a narrow wave."
-		update_icon()
-		return TRUE
-	return ..()
+	. = ..()
+	icon_state = bomb ? icon_state_loaded : initial(icon_state)
+
+/obj/item/gun/blastcannon/attackby(obj/item/transfer_valve/_bomb, mob/user)
```

Comment:
I don't like using `_name` to refer to things that aren't unused, change this to `new_bomb` or similar.

Reply:
Or just name it `bomb` and do `src.bomb = bomb` at the end, I'm quite liking that style.